program-runtime/src/invoke_context.rs

```diff
@@ -1154,18 +1154,14 @@ pub fn mock_process_instruction(
 }
 
 /// Visit each unique instruction account index once
-fn visit_each_account_once(
+pub fn visit_each_account_once<E>(
     instruction_accounts: &[InstructionAccount],
-    work: &mut dyn FnMut(usize, &InstructionAccount) -> Result<(), InstructionError>,
-) -> Result<(), InstructionError> {
+    work: &mut dyn FnMut(usize, &InstructionAccount) -> Result<(), E>,
+) -> Result<(), E> {
     'root: for (index, instruction_account) in instruction_accounts.iter().enumerate() {
         // Note: This is an O(n^2) algorithm,
         // but performed on a very small slice and requires no heap allocations
-        for before in instruction_accounts
-            .get(..index)
-            .ok_or(InstructionError::NotEnoughAccountKeys)?
-            .iter()
-        {
+        for before in instruction_accounts.get(..index).unwrap().iter() {
```

Comment:
Can't do `.ok_or(InstructionError::NotEnoughAccountKeys)` here anymore because of the generic result type. But this can never fail as `index` which is used in `instruction_accounts.get()` was acquired from `instruction_accounts.iter().enumerate()`.

Reply:
How about this: https://github.com/solana-labs/solana/pull/23701/commits/aac309b283e0463e683077d722646f6566632d9d